plugins/pcl/filters/HeightFilter.cpp

```diff
@@ -0,0 +1,182 @@
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "HeightFilter.hpp"
+
+// c++
+#include <memory>
+#include <string>
+#include <vector>
+
+// project
+#include "PCLConversions.hpp"
+#include <pdal/Dimension.hpp>
+#include <pdal/Options.hpp>
+#include <pdal/pdal_macros.hpp>
+#include <pdal/PointTable.hpp>
+#include <pdal/PointView.hpp>
+#include <pdal/StageFactory.hpp>
+
+// other
+#include <pcl/console/print.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/filters/project_inliers.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/point_types.h>
+#include <pcl/search/kdtree.h>
+
+#define DBG log()->get(LogLevel::Debug)
+
+namespace pdal
+{
+
+typedef pcl::PointCloud<pcl::PointXYZ> Cloud;
+
+static PluginInfo const s_info =
+    PluginInfo("filters.height", "Height Filter", "");
+
+CREATE_SHARED_PLUGIN(1, 0, HeightFilter, Filter, s_info)
+
+std::string HeightFilter::getName() const
+{
+    return s_info.name;
+}
+
+Options HeightFilter::getDefaultOptions()
+{
+    Options options;
+    return options;
+}
+
+void HeightFilter::addDimensions(PointLayoutPtr layout)
+{
+    m_heightDim = layout->registerOrAssignDim("Height", Dimension::Type::Double);
+}
+
+void HeightFilter::filter(PointView& view)
+{
+    if (!view.hasDim(Dimension::Id::Classification))
```

Comment:
This check is better in prepared(), as it will get called before any execution starts and points are read.